crates/sui-types/src/message_envelope.rs

```diff
@@ -257,6 +257,10 @@ impl<T: Message, S> VerifiedEnvelope<T, S> {
         &self.0 .0
     }
 
+    pub fn inner_mut(&mut self) -> &mut Envelope<T, S> {
+        &mut self.0 .0
+    }
```

Comment:
This breaks the VerifiedEnvelope invariants and needs to be removed. Having access to an &mut would let code completely replace the inner type without verification. cc @mystenmark @lxfind